### PR TITLE
Import turtle #220

### PR DESCRIPTION
**Pull request by JervenBolleman**
This allows us to import RDF VG graphs. i.e. roundtrip them back and forth.

### CODE REVIEW

File: src/vg.cpp [hunk 1]
```diff
@@ -2162,6 +2181,124 @@ void VG::from_gfa(istream& in, bool showp) {
 		}
 
 }
+    static
+    void
+    triple_to_vg(void* user_data, raptor_statement* triple)
+    {
+        VG* vg = ((std::pair<VG*, Paths*>*) user_data)->first;
+        Paths* paths = ((std::pair<VG*, Paths*>*) user_data)->second;
+        const string vg_ns ="<http://example.org/vg/";
```
Comment (macieksmuga) [inline src/vg.cpp]: Can we change this now to `http://github.com/vgteam/vg`, or is there more magic involved in preparing such a URI prefix before we can set it thusly?

Comment (JervenBolleman) [inline src/vg.cpp]: No magic at all. But best to pick uri to be decided in pull vg#217

In src/vg.cpp <https://github.com/vgteam/vg/pull/233#discussion_r54221005>:

> @@ -2162,6 +2181,124 @@ void VG::from_gfa(istream& in, bool showp) {
>  		}
>
>  }
> +    static
> +    void
> +    triple_to_vg(void* user_data, raptor_statement* triple)
> +    {
> +        VG* vg = ((std::pair<VG*, Paths*>*) user_data)->first;
> +        Paths* paths = ((std::pair<VG*, Paths*>*) user_data)->second;
> +        const string vg_ns ="<http://example.org/vg/";

Can we change this now to http://github.com/vgteam/vg, or is there more
magic involved in preparing such a URI prefix before we can set it thusly?

—
Reply to this email directly or view it on GitHub
<https://github.com/vgteam/vg/pull/233/files#r54221005>.

**Comment by JervenBolleman** [comment]
Closing this one to make nicer pull request with the same funcionality

**Comment by macieksmuga** [comment]
Bummer, I just got around to testing it (sorry so late). The only issue I had was that if I tried to export from VG to turtle with `view -t`, then import back in with `view -T`, the pipe character in identifiers wasn't being properly escaped (There may also be an outstanding issue with hash (`#`) characters also not being escaped properly when creating turtle files):
```
➜  maciek-vg git:(JervenBolleman-220_import_turtle.1) ✗ vg view -V -r http://github.com/ekg/vg -t ../../Sendai\ Biohackathon/B-3106.vg > b3.ttl

➜  maciek-vg git:(JervenBolleman-220_import_turtle.1) ✗ vg view -v -r http://github.com/ekg/vg -T b3.ttl > b3.vg
URI http://github.com/ekg/vg:646 raptor error - syntax error at '|'
```
(the `B-3106.vg` file was created with the `vg msga` algorithm, pointed at `test/GRCh38_alts/FASTA/HLA/B-3106.fa`).

**Pull request closed without merging**